unittests/savanna_misc_tests.cpp

```diff
@@ -688,5 +687,106 @@ BOOST_FIXTURE_TEST_CASE(verify_spring_1_0_block_compatibitity, savanna_cluster::
+      fc::scoped_set_value tmp_C(C.vote_delay(), 1);
+
+      b1 = A.produce_block();
+      print("b1", b1);
+      BOOST_REQUIRE_EQUAL(qc_s(qc(b1)), strong_qc(b0)); // b1 claims a strong QC on b0
+
+      b2 = A.produce_block();
+      print("b2", b2);
+      BOOST_REQUIRE(!qc(b2));                           // b2 should not include a QC (votes on b1 delayed)
+   }
+
+   B.propagate_delayed_votes_to(D);                     // propagate votes on b2 to D, so it can form a QC on b2
+   C.propagate_delayed_votes_to(D);                     // which will be included in b6
+
+   const std::vector<size_t> partition {3};             // partition D so that it doesn't see b3, b4 and b5
+   set_partition(partition);                            // and don't vote on it
+
+   auto b3 = A.produce_block();
+   print("b3", b3);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b3)), strong_qc(b1));    // b3 claims a strong QC on b1 (votes on b2 delayed)
+
+   D.push_block(b3);                                    // we want D to see b3, but not receive the votes on
+                                                        // b3, so that when it produces b6, b6 will have a
+                                                        // qc claim on b2
+
+   auto b4 = A.produce_block();
+   print("b4", b4);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b4)), strong_qc(b3));    // b4 claims a strong QC on b3
+
+   auto b5 = A.produce_block();
+   print("b5", b5);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b5)), strong_qc(b4));    // b5 claims a strong QC on b4
+
+
+   set_partition({});                                   // remove partition so all nodes can vote on b6 and above
+
+   auto b6 = D.produce_block(_block_interval_us * 3);   // D (who has not seen b4 and b5) produces b6
+                                                        // b6 has a higher timestamp than b5
+   print("b6", b6);
+   BOOST_REQUIRE_EQUAL(b6->previous, b3->calculate_id());
+   BOOST_REQUIRE(!!qc(b6));                             // b6 should include a QC
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b6)), strong_qc(b2));    // b6 claims a strong QC on b2
+
+   // ---------------------------------------------------------------------------------------------------
+   // After voting on `b5` (which makes `b3` final), the finalizers who voted on `b5` are locked on `b4`,
+   // and therefore cannot vote on `b6`:
+   //
+   // - liveness check fails because: `b6' s core.latest_qc_block_timestamp() <  fsi.lock.timestamp`
+   //   because `b2 timestamp < b4 timestamp`.
+   // - safety check fails because `b6` does not extend `b4`.
+   //
+   // As a result, we don't expect the next block (b7) to include a QC
+   // ---------------------------------------------------------------------------------------------------
+
+   auto b7 = D.produce_block();                         // D produces a block. It still has not seen b4 and b5.
+   print("b7", b7);
+   BOOST_REQUIRE(!qc(b7));                              // b7 should include a QC
```

Comment:
Comment should be `b7 should not include a QC`, right?

Reply:
https://github.com/AntelopeIO/spring/pull/782/commits/78c1053e75ccde144fc424429c38e4800d760aa9